src/python/txtai/pipeline/text/translation.py

```diff
@@ -176,10 +184,11 @@ def translate(self, texts, source, target):
         # Combine translations - handle splits on large text from tokenizer
         results, last = [], -1
         for x, i in enumerate(indices):
+            v = (model, translated[x]) if debug else translated[x]
```

Comment:
I think it would be best to return the model path vs the PyTorch model here. In other words, this can be a string. That way if this method were to be called via the API, it would still work.

Reply:
I see, should i use
```
model.__class__.__name__
```
or 

```
model.config.name_or_path
```

Not sure what is the correct way to get the model name

Reply:
I think the cleanest thing to do would be to return the path with the lookup method.

```python
    def lookup(self, source, target):
        """
        Retrieves a translation model for source->target language. This method caches each model loaded.

        Args:
            source: source language code
            target: target language code

        Returns:
            (path, model, tokenizer) [CHANGED]
        """

        # Determine best translation model to use, load if necessary and return
        path = self.modelpath(source, target)
        if path not in self.models:
            self.models[path] = self.load(path)

        return (path,) + self.models[path] [CHANGED]
```

Then you would change the line:

```python
        # Load model and tokenizer
        path, model, tokenizer = self.lookup(source, target)
```

Reply:
Thank you for these details !

Reply:
I have made these changes

Reply:
Looks good!